vizro-core/src/vizro/models/_components/table.py

```diff
@@ -51,12 +54,36 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    def _build_underlying_table_object(self):
+        data = data_manager._get_component_data(self.id)  # type: ignore
+        kwargs = self.figure._arguments.copy()
+        kwargs.pop("data_frame", None)
+        return self.figure._function(data_frame=data, **kwargs)
+
+    @_log_call
+    def pre_build(self):
+        if self.actions:
+            # The underlying table object is pre-built, so we can fetch its ID.
+            underlying_table_object = self._build_underlying_table_object()
+
+            # Underlying table object has to have "id" defined if it triggers actions chain.
+            if not hasattr(underlying_table_object, "id"):
+                raise ValueError(
+                    "Underlying table object has no attribute 'id'. To perform specified actions, a valid 'id' to the"
+                    " Underlying table object has to be provided."
+                )
+
+            self._underlying_table_id = underlying_table_object.id
+
     @_log_call
     def build(self):
+        # TODO: We also need to take case what empty object we need to create here once we support AgGrid and others.
         return html.Div(
             [
                 html.H3(self.title, className="table-title") if self.title else None,
-                html.Div(dash_table.DataTable(), id=self.id),
+                html.Div(
+                    dash_table.DataTable(**({"id": self._underlying_table_id} if self.actions else {})), id=self.id
+                ),
```

Comment:
After we support AgGrid and others, we will also need to take care of the empty object we create here.